.github/workflows/quality-tests.yml

```diff
@@ -0,0 +1,81 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+
+jobs:
+  run_tests:
+    name: Quality
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04 ]
+        python-version: [ 3.8 ]
+        node-version: [ 12 ]
+        shard: [ 1, 2, 3, 4 ]
+
+    steps:
+
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+
+    - name: Fetch master for comparison
+      run: git fetch --depth=1 origin master
+
+    - name: Install Required System Packages
+      run: sudo apt-get update && sudo apt-get install libxmlsec1-dev
+
+    - name: Setup Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Setup Node
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node-version }}
+
+
+    - name: Get pip cache dir
+      id: pip-cache-dir
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: Cache pip dependencies
+      id: cache-dependencies
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.pip-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements/edx/development.txt') }}
+        restore-keys: ${{ runner.os }}-pip-
+
+    - name: Install Required Python Dependencies
+      env:
+        PIP_SRC_DIR: ${{ runner.temp }}
+      run: |
+        pip install -r requirements/pip.txt
+        pip install -r requirements/edx/development.txt --src $PIP_SRC_DIR
+
+    - name: Run Quality Tests
+      env:
+        TEST_SUITE: quality
+        SCRIPT_TO_RUN: ./scripts/generic-ci-tests.sh
```

Comment:
why don't we instead use quality commands directly in workflow, it'd be a lot cleaner and will allow us to cleanup those scripts eventually